### PR TITLE
correct conditionals in `component`

### PR DESCRIPTION
**Pull request by bekzod**

### CODE REVIEW

**Review by rwjblue — COMMENTED** [review]
Does this fix any specific thing? Can we add a regression test for it?

File: packages/ember-glimmer/lib/component.ts [hunk 1]
```diff
@@ -604,10 +604,8 @@ const Component = CoreView.extend(
       let args = this[ARGS];
       let reference = args && args[key];
 
-      if (reference) {
-        if (reference[UPDATE]) {
-          reference[UPDATE](get(this, key));
-        }
+      if (reference && UPDATE in reference) {
```
Comment (krisselden) [inline packages/ember-glimmer/lib/component.ts]: lets not use then, use reference[UPDATE] !== undefined

Comment (krisselden) [reply packages/ember-glimmer/lib/component.ts]: in general we should avoid falsy conversions

**Review by krisselden — CHANGES_REQUESTED** [review]

**Comment by krisselden** [comment]
@rwjblue he is just collapsing the if, there isn't change to behavior

File: packages/ember-glimmer/lib/component.ts [hunk 2]
```diff
@@ -602,12 +602,10 @@ const Component = CoreView.extend(
...
 
       let args = this[ARGS];
-      let reference = args && args[key];
+      let reference = args !== undefined ? args[key] : undefined;
 
-      if (reference) {
-        if (reference[UPDATE]) {
-          reference[UPDATE](get(this, key));
-        }
+      if (reference !== undefined && UPDATE in reference) {
```
Comment (rwjblue) [inline packages/ember-glimmer/lib/component.ts]: Can you update this to:

```js
if (reference !== undefined && reference[UPDATE] !== undefined) {
}
```

(no need to use `in` here AFAICT)

**Review by rwjblue — CHANGES_REQUESTED** [review]

**Review by krisselden — APPROVED** [review]

**Review by rwjblue — APPROVED** [review]

**Pull request merged**